src/cron.php

```diff
@@ -34,27 +35,46 @@
     return compact('cachet', 'pingdom');
 };
 
+function write($line)
+{
+    echo $line . PHP_EOL;
+}
+
 // Parse the metrics & components map
 $metricsMap    = array_map($extractMap, explode(',', getenv('METRICS_MAP')));
 $componentsMap = array_map($extractMap, explode(',', getenv('COMPONENTS_MAP')));
 
 // Initialize the Cachet client library
-$cachetClient = new CachetClient(getenv('CACHET_HOST') . '/api/v1/', getenv('CACHET_API_KEY'));
-$cachetPoints = PointFactory::build($cachetClient);
+$cachetClient     = new CachetClient(getenv('CACHET_HOST') . '/api/v1/', getenv('CACHET_API_KEY'));
+$componentManager = ComponentFactory::build($cachetClient);
+$cachetPoints     = PointFactory::build($cachetClient);
 
 // Initialize the Pingdom client library
 $pingdomClient = new Client(getenv('PINGDOM_USERNAME'), getenv('PINGDOM_PASSWORD'), getenv('PINGDOM_API_KEY'));
 
-// Update the components
-// @TODO(stayallive): Implement the component updates
+$checks = $pingdomClient->getChecks();
+foreach ($checks as $check) {
+    foreach ($componentsMap as $componentMap) {
+        if ($componentMap['pingdom'] == $check['id']) {
+            write("[Component] Updating Pingdom {$componentMap['pingdom']} to Cachet {$check['id']} with status: {$check['status']}");
+
+            $component = $componentManager->updateComponent($componentMap['cachet'], [
+                'status' => ($check['status'] == 'up' ? 1 : 4),
+            ]);
+        }
+    }
+}
 
 // Update the metrics
 foreach ($metricsMap as $metricMap) {
-    $result = $pingdomClient->getResults($metricMap['pingdom'], 1)[0];
-    $point  = ['value' => $result['responsetime'], 'timestamp' => $result['time']];
+    $results = $pingdomClient->getResults($metricMap['pingdom'], 2);
```

Comment:
Grabbing a second metric point allows padding for the seconds that Pingdom can delay a check. Since Cachet understands _when_ a metric point happened, they can be smoothed out.